front-app/src/views/Products.tsx

```diff
@@ -0,0 +1,188 @@
+/* eslint-disable react/jsx-key */
+import React, { useEffect, useState } from 'react';
+
+import InfiniteScroll from 'react-infinite-scroller';
+import { useQuery } from 'react-query';
+
+import { getCategories } from '../api/categoriesApi';
+import { getProducts } from '../api/productsApi';
+import SearchIcon from '../assets/svg/search_icon.svg';
+import Vector from '../assets/svg/vector.svg';
+import HelpComponent from '../components/HelpComponent';
+import ProductCard from '../components/ProductCard';
+import TopNav from '../layouts/TopNav';
+
+interface Product {
+  id: number;
+  name: string;
+  price: number;
+  image: string;
+  categoryId: number;
+  createdAt: string;
+  updatedAt: string;
+}
+
+interface Category {
+  id: number;
+  name: string;
+  createdAt: string;
+  updatedAt: string;
+}
+
+const Products = () => {
+  const [queryName, setQueryName] = useState<null | string>(null);
+  const [queryCategories, setQueryCategories] = useState<null | number[]>(null);
+  const [querySortBy, setQuerySortBy] = useState('');
+  const [queryPage, setQueryPage] = useState(0);
+  const [querySize, setQuerySize] = useState(4);
+  const [productList, setProductToList] = useState<Product[]>([]);
+  const [hasMoreValue, setHasMore] = useState(true);
+
+  const filters = {
+    name: queryName,
+    categories: queryCategories,
+    sortBy: querySortBy,
+    page: queryPage,
+    size: querySize,
+  };
+
+  const { isLoading, data } = useQuery<{ products: Product[]; totalCount: number }>(['params', filters], () =>
+    getProducts(filters),
+  );
+  const { data: categoriesData } = useQuery<Category[]>(['params'], () => getCategories());
+
+  const totalResults = data?.totalCount;
+
+  const handleSortBy = (event: any) => {
+    setQuerySortBy(event.target.value);
+  };
+
+  const [textValue, setTextValue] = useState('');
+  const handleInputChange = (event: any) => {
+    setTextValue(event.target.value);
+  };
+  const handleInputClick = () => {
+    setQueryName(textValue);
+  };
+
+  const [categoryFilters, setFilters] = useState<number[]>([]);
+  const handleCategoryClick = (category: number) => {
+    const index = categoryFilters.indexOf(category);
+    index === -1
+      ? setFilters([...categoryFilters, category])
+      : setFilters([...categoryFilters.slice(0, index), ...categoryFilters.slice(index + 1)]);
+  };
+  useEffect(() => {
+    setQueryCategories(categoryFilters);
+  }, [categoryFilters]);
+
```

Comment:
Let's refactor this parts of the code. The standards that we usually follow are having all the useStates all together, and then the same for the effects, and then all your custom function together as well. Apart from that let's add some more spacing between the effects just to make the code more readable.